db tools: Rename "rebuild-" tools and improve digest schemes.

showell opened the pull request:
This should have user-facing improvements, and it also cleans up a good chunk of code, either removing code outright or simplifying conditional logic.

There is gonna be a one-time forced rebuilding of the databases to introduce the new digest scheme.

showell commented:
@timabbott If you merge this, please do a little bump on this Zulip thread: https://chat.zulip.org/#narrow/stream/3-backend/topic/rebuilding.20databases/near/860130

As you'll see there, I just try to give folks a bit of a heads up on the one-time changes.

timabbott commented on zerver/lib/test_fixtures.py:
```diff
@@ -104,29 +132,27 @@ def database_exists(self) -> bool:
             return False
 
-    def is_current(self) -> bool:
+    def needs_rebuild(self) -> bool:
```
I think this might be better named `database_inputs_changed` -- we have another `needs_rebuild` case from migrations having been removed, below, that it's worth distinguishing this from.

showell replied:
I went with `files_or_settings_have_changed`, but if you prefer `database_inputs_changed`, please just change it for me.   (I ended up including settings in the new push.)

timabbott commented on zerver/lib/test_fixtures.py:
```diff
@@ -123,18 +121,12 @@ def is_current(self) -> bool:
         ]
-        check_settings = [
-            'REALM_INTERNAL_BOTS',
-        ]
```
I don't want to remove the `check_settings` infrastructure; we're definitely going to end up needing to re-add it in the future.  (And I think there are probably a few settings that impact `populat_db` and probably we'll add to this list the next time we change them).

I like the direction of the refactor, though; maybe we can just add a new argument to `is_digest_obsolete` that's a dictionary (extracted from `settings`) of custom values to hash a stable marshalling (sort-then-json-dump?) together with the rest?

showell replied:
Based on your input, I kept `REALM_INTERNAL_BOTS` and added two other bot-related settings which are now more likely to trigger a rebuild.

timabbott commented:
I merged all but the last 3 commits, and posted a few comments.

showell commented:
I have re-pushed.  I think the two new commits address your earlier concerns.
The first commit is brand new to you, and it's all renaming and comments.

Then I dropped the drop-setting commit.

Finally I squashed two intermediate commits that didn't really make the PR easier to read and which were sort of local minima, even though they weren't truly broken.
